ambry-router/src/main/java/com.github.ambry.router/GetManager.java

```diff
@@ -252,3 +276,32 @@ private void removeAndAbort(GetOperation op, Exception abortCause) {
   }
 }
 
+/**
+ * Extension to {@link GetBlobOptions} containing additional parameters to the GetBlob operation that is internal to the
+ * router.
+ */
+class ExtendedGetBlobOptions extends GetBlobOptions {
```

Comment:
`GetBlobOptions` is under ambry-api and is exposed to externals. We don't usually have private constructors and such in these classes and I am not very comfortable with that either. It is not the right place to include things internal to the router - and going forward the router may want to pass around more "options" internally. The other choice was to pass in yet another parameter to the methods taking the `GetBlobOptions`, but I felt that having an internal `Options` class would help in the future as well. 

Having a builder is orthogonal to all this.